java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -478,41 +429,39 @@ private Resolution findMethod(Env env, JavaType callSite, JavaType site, String
     return bestSoFar;
   }
 
-  private JavaType handleTypeArguments(List<JavaType> typeParams, JavaType type, JavaSymbol.MethodJavaSymbol methodSymbol) {
-    if (!typeParams.isEmpty() && methodSymbol.typeVariableTypes.size() == typeParams.size()) {
-      TypeSubstitution typeSubstitution = new TypeSubstitution();
-      int i = 0;
-      for (JavaType.TypeVariableJavaType typeVariableType : methodSymbol.typeVariableTypes) {
-        typeSubstitution.add(typeVariableType, typeParams.get(i));
-        i++;
-      }
-      return substituteTypeParameter(type, typeSubstitution);
-    }
-    return type;
+  private static boolean isConstructor(JavaSymbol symbol) {
+    return "<init>".equals(symbol.name);
   }
 
   /**
    * @param candidate    candidate
    * @param bestSoFar previously found best match
    */
-  private JavaSymbol selectBest(Env env, JavaType site, List<JavaType> argTypes, JavaSymbol candidate, JavaSymbol bestSoFar, boolean autoboxing) {
+  private JavaSymbol selectBest(Env env, JavaType site, List<JavaType> argTypes, List<JavaType> typeParams, JavaSymbol candidate, JavaSymbol bestSoFar, boolean autoboxing) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1142)